client/src/views/client-area/DashboardPage.vue

```diff
@@ -1,18 +1,520 @@
+          @click="switchPage(ClientAreaPages.PaymentMethods)"
+        >
+          {{ $msTranslate('clientArea.dashboard.payment.update') }}
+        </ion-text>
+      </div>
+    </div>
   </div>
 </template>
 
 <script setup lang="ts">
-import { BmsOrganization } from '@/services/bms';
+import { IonTitle, IonText, IonList, IonItem, IonIcon, IonSkeletonText } from '@ionic/vue';
+import { download } from 'ionicons/icons';
+import { ClientAreaPages } from '@/views/client-area/types';
+import {
+  BmsAccessInstance,
+  DataType,
+  BmsOrganization,
+  OrganizationStatsResultData,
+  BmsInvoice,
+  BillingDetailsPaymentMethodCard,
+  PaymentMethod,
+} from '@/services/bms';
+import { MsInformationTooltip, I18n, MsStripeCardDetails } from 'megashark-lib';
+import { onMounted, ref } from 'vue';
+import { PaymentMethod as MsPaymentMethod } from 'megashark-lib';
+import { navigateTo, Routes } from '@/router';
+import { DateTime } from 'luxon';
 
-defineProps<{
+const props = defineProps<{
   organization: BmsOrganization;
 }>();
+
+const stats = ref<OrganizationStatsResultData | undefined>(undefined);
+const invoices = ref<Array<BmsInvoice>>([]);
+const defaultCard = ref<MsPaymentMethod.Card | undefined>(undefined);
+const currentDate = DateTime.now();
+
+// should be fetched from the backend
+const ADMIN_USER_PRICE = 15;
+
+onMounted(async () => {
+  const orgStatsResponse = await BmsAccessInstance.get().getOrganizationStats(props.organization.bmsId);
+  if (!orgStatsResponse.isError && orgStatsResponse.data && orgStatsResponse.data.type === DataType.OrganizationStats) {
+    stats.value = orgStatsResponse.data;
+  }
+
+  const invoicesResponse = await BmsAccessInstance.get().getInvoices();
+  if (!invoicesResponse.isError && invoicesResponse.data && invoicesResponse.data.type === DataType.Invoices) {
+    invoices.value = invoicesResponse.data.invoices.filter((invoice) => invoice.organizationId === props.organization.parsecId);
+  }
+
+  const billingDetailsResponse = await BmsAccessInstance.get().getBillingDetails();
+  if (!billingDetailsResponse.isError && billingDetailsResponse.data && billingDetailsResponse.data.type === DataType.BillingDetails) {
+    const card = billingDetailsResponse.data.paymentMethods.find(
+      (method) => method.isDefault && method.type === PaymentMethod.Card,
+    ) as BillingDetailsPaymentMethodCard;
+    if (card) {
+      defaultCard.value = {
+        last4: card.lastDigits,
+        // eslint-disable-next-line camelcase
+        exp_year: card.expirationDate.year,
+        // eslint-disable-next-line camelcase
+        exp_month: card.expirationDate.month,
+        brand: card.brand,
+      } as MsPaymentMethod.Card;
+    }
+  }
+});
+
+async function switchPage(page: ClientAreaPages): Promise<void> {
+  navigateTo(Routes.ClientArea, { skipHandle: true, query: { organization: props.organization.bmsId, page: page } });
+}
+
+// temporary function, should be fetched from the backend
```

Comment:
Add a big todo `// TODO: XXX` and open a issue.

Reply:
Yup this is a big nope, the front-end should not do this, because if we change the formula or have multiple in the future, the front-end will not be able to keep up and it will duplicate every formula in the 2 projects.
So, basically don't do this, it's the BMS role, open an issue on the BMS repo to add a route for this.